x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/job/DateHistogramGroupConfig.java

```diff
@@ -68,9 +93,82 @@
     private final DateHistogramInterval delay;
     private final String timeZone;
 
+    /**
+     * FixedInterval is a {@link DateHistogramGroupConfig} that uses a fixed time interval for rolling up data.
+     * The fixed time interval is one or multiples of SI units and has no calendar-awareness (e.g. doesn't account
+     * for leap corrections, does not have variable length months, etc).
+     *
+     * For calendar-aware rollups, use {@link CalendarInterval}
+     */
+    public static class FixedInterval extends DateHistogramGroupConfig {
+        public FixedInterval(String field, DateHistogramInterval interval) {
+            this(field, interval, null, null);
+        }
+
+        public FixedInterval(String field, DateHistogramInterval interval, DateHistogramInterval delay, String timeZone) {
+            super(field, interval, delay, timeZone);
+            // validate fixed time
+            TimeValue fixedInterval = TimeValue.parseTimeValue(interval.toString(), NAME + ".FixedInterval");
+        }
+
+        FixedInterval(StreamInput in) throws IOException {
+            super(in);
+        }
+    }
+
+    /**
+     * CalendarInterval is a {@link DateHistogramGroupConfig} that uses calendar-aware intervals for rolling up data.
+     * Calendar time intervals understand leap corrections and contextual differences in certain calendar units (e.g.
+     * months are variable length depending on the month).  Calendar units are only available in singular quantities:
+     * 1s, 1m, 1h, 1d, 1w, 1q, 1M, 1y
+     *
+     * For fixed time rollups, use {@link FixedInterval}
+     */
+    public static class CalendarInterval extends DateHistogramGroupConfig {
+        public CalendarInterval(String field, DateHistogramInterval interval) {
+            this(field, interval, null, null);
+
+        }
+
+        public CalendarInterval(String field, DateHistogramInterval interval, DateHistogramInterval delay, String timeZone) {
+            super(field, interval, delay, timeZone);
+            if (DateHistogramAggregationBuilder.DATE_FIELD_UNITS.get(interval.toString()) == null) {
+                throw new IllegalArgumentException("The supplied interval [" + interval +"] could not be parsed " +
+                    "as a calendar interval.");
+            }
+        }
+
+        CalendarInterval(StreamInput in) throws IOException {
+            super(in);
+        }
+    }
+
+    static DateHistogramGroupConfig fromUnknownTimeUnit(String field, DateHistogramInterval interval,
```

Comment:
Could you add a JavaDocs to this and the method below explaining when this should be used and when the normal constructors should be used? Its not clear to me when it would be appropriate to use the normal constructors?

Reply:
Sure.  The tl;dr: is this handles the legacy case where jobs are stored with `interval` and we need to "up-convert" that to one of fixed/calendar.  Once the deprecation period is over, we'll remove the default ctor so a user can only choose Fixed or Calendar directly (in java), and remove the option for `interval` in the REST layer.